### PR TITLE
chat page commit

### PR DESCRIPTION
**Pull request by nester7x**
image

### CODE REVIEW

File: src/components/Layout/index.jsx [hunk 1]
```diff
@@ -48,14 +49,23 @@ const Layout = ({ children }) => {
...
+    },
     {
       component: (
         <S.Link to="users">
           <GroupIcon />
         </S.Link>
       ),
       isLogin: 'loggedIn',
-      key: 4
+      key: 5
```
Comment (SerhiienkoRuslan) [inline src/components/Layout/index.jsx]: use "users" for key

File: src/context/global.jsx [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
+    userInfo: {},
+    isLogin: false
+  });
 
   useEffect(() => {
-    const token = getCookie('token');
-    if (token) {
-      setLogin(true);
-    }
+    const putData = async () => {
```
Comment (SerhiienkoRuslan) [inline src/context/global.jsx]: putData rename to getUser

File: src/screens/Chat/components/Messages/index.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from 'react';
+import * as S from './styles';
+import Message from '../Message';
+import { httpGet } from '../../../../api/base.api';
+import { getCookie } from '../../../../utils/CookieUtils';
```
Comment (SerhiienkoRuslan) [inline src/screens/Chat/components/Messages/index.jsx]: use absolute path (setup needed)
'../../../../utils/CookieUtils' => 'utils/CookieUtils'

File: src/screens/Chat/components/SendMessage/index.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  const sendMessage = async () => {
+    const token = getCookie('token');
+
+    const data = await httpPost(
+      'message',
+      {
+        from: user?.userInfo?.email,
+        to:
```
Comment (SerhiienkoRuslan) [inline src/screens/Chat/components/SendMessage/index.jsx]: ??

Comment (nester7x) [reply src/screens/Chat/components/SendMessage/index.jsx]: do you mean what does this line mean?

File: src/components/Layout/index.jsx [hunk 5]
```diff
@@ -82,10 +92,10 @@ const Layout = ({ children }) => {
...
               }
-              if (isLogin && item.isLogin === 'loggedIn') {
+              if (user.isLogin && item.isLogin === 'loggedIn') {
                 return <li key={item.key}>{item.component}</li>;
               }
-              if (!isLogin && item.isLogin === 'loggedOut') {
+              if (!user.isLogin && item.isLogin === 'loggedOut') {
                 return <li key={item.key}>{item.component}</li>;
               }
               return '';
```
Comment (aspasionov) [inline src/components/Layout/index.jsx]: instead of this, make function with 'swich case'

**Comment by nester7x** [comment]
text comment

**Comment by nester7x** [comment]
image

**Comment by nester7x** [comment]
image
image

File: src/screens/Chat/components/Messages/index.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import { httpGet } from 'api/base.api';
+import { getCookie } from 'utils/CookieUtils';
+import Message from '../Message';
+import * as S from './styles';
+
+const Messages = () => {
+  const { receiver, user } = useContext(GlobalContext);
+  const [info, setInfo] = useState([]);
+
+  useEffect(() => {
```
Comment (aspasionov) [inline src/screens/Chat/components/Messages/index.jsx:14]: can show preloader during loading?

File: src/screens/Registration/index.jsx [hunk 7]
```diff
@@ -29,48 +42,85 @@ const Registration = () => {
     window.location.reload();
```
Comment (aspasionov) [inline src/screens/Registration/index.jsx:42]: remove this,
submit message without reloading page

File: src/context/global.jsx [hunk 8]
```diff
@@ -13,6 +13,11 @@ export const GlobalContextProvider = ({ children }) => {
     isLogin: false
   });
 
+  const [receiver, setReceiver] = useState({
```
Comment (SerhiienkoRuslan) [inline src/context/global.jsx]: it shouldn't be in global context

File: src/screens/Chat/index.jsx [hunk 9]
```diff
@@ -1,34 +1,43 @@
...
+        const data = await httpGet(`user/${id}`, getCookie('token'));
+        await setReceiverName(data.user.username);
+      };
+      setReceiver((prevState) => ({
+        ...prevState,
+        receiverId: id
+      }));
 
-    getData();
+      getData();
```
Comment (SerhiienkoRuslan) [inline src/screens/Chat/index.jsx]: use IIFE instead getData

(async () {
  await ....
})()

**Review by SerhiienkoRuslan — APPROVED** [review]

**Review by aspasionov — APPROVED** [review]

**Review by SerhiienkoRuslan — APPROVED** [review]

**Pull request merged**